addons/block_code/ui/blocks/block/block.gd

```diff
@@ -214,7 +214,40 @@ func _to_string():
 
 
 func _get_tooltip(at_position: Vector2) -> String:
-	return definition.description if definition else ""
+	var variant_as_string: String
+
+	if definition:
+		if definition.variant_type == Variant.Type.TYPE_NIL:
+			return definition.description
+		elif definition.variant_type == Variant.Type.TYPE_BOOL:
+			variant_as_string = "Boolean"
+		elif definition.variant_type == Variant.Type.TYPE_INT:
+			variant_as_string = "Integer"
+		elif definition.variant_type == Variant.Type.TYPE_FLOAT:
+			variant_as_string = "Float"
+		elif definition.variant_type == Variant.Type.TYPE_STRING:
+			variant_as_string = "String"
+		elif definition.variant_type == Variant.Type.TYPE_VECTOR2:
+			variant_as_string = "Vector2"
+		elif definition.variant_type == Variant.Type.TYPE_COLOR:
+			variant_as_string = "Color"
+		elif definition.variant_type == Variant.Type.TYPE_NODE_PATH:
+			variant_as_string = "NodePath"
+		elif definition.variant_type == Variant.Type.TYPE_OBJECT:
+			variant_as_string = "Object"
+		elif definition.variant_type == Variant.Type.TYPE_CALLABLE:
+			variant_as_string = "Callable"
+		elif definition.variant_type == Variant.Type.TYPE_SIGNAL:
+			variant_as_string = "Signal"
+		elif definition.variant_type == Variant.Type.TYPE_DICTIONARY:
+			variant_as_string = "Dictionary"
+		elif definition.variant_type == Variant.Type.TYPE_ARRAY:
+			variant_as_string = "Array"
+		else:
+			variant_as_string = "Undefined"
+
+		return definition.description + "\n\nType: [b]" + variant_as_string + "[/b]"
```

Comment:
Can you instead do this:

```suggestion
		return definition.description + "\n\nType: [b]" + type_string(definition.variant_type)
```

Reply:
except perhaps not appending the type in the NIL case, as you already did. I'll edit my suggestion

Reply:
Thanks for amending this. When I got back to my computer to test the branch I thought it would be clearer to use a format string, so I took the liberty of making that change in your patch.